lib/cli/run-helpers.js

```diff
@@ -196,14 +196,19 @@ exports.singleRun = (mocha, {files = [], exit = false} = {}) => {
   mocha.files = files;
   const runner = mocha.run(exit ? exitMocha : exitMochaLater);
 
-  process.on('SIGINT', () => {
-    debug('aborting runner');
-    runner.abort();
+  process.on('SIGINT', function forceStop() {
+    if (forceStop.lock === undefined) {
+      forceStop.lock = true;
+      debug('aborting runner');
+      runner.abort();
 
-    // This is a hack:
-    // Instead of `process.exit(130)`, set runner.failures to 130 (exit code for SIGINT)
-    // The amount of failures will be emitted as error code later
-    runner.failures = 130;
+      // This is a hack:
+      // Instead of `process.exit(130)`, set runner.failures to 130 (exit code for SIGINT)
+      // The amount of failures will be emitted as error code later
+      runner.failures = 130;
+      // Reraising the signal
+      setImmediate(() => process.kill(process.pid, 'SIGINT'));
```

Comment:
you don't need `lock`, just stop listening:

```js
process.on('SIGINT', function forceStop() {
  debug('aborting runner');
  runner.abort();
  runner.failures = 130;
  setImmediate(() => process.kill(process.pid, 'SIGINT'));
  process.removeListener('SIGINT', forceStop);
});
```

Reply:
or even `process.once`!

Reply:
Updated as per the suggestion